main.py

```diff
@@ -409,6 +416,62 @@ def check_arg_name(args):
         )
 
 
+def reload_model(args, ckpt_path, model, index, strict, task):
+    """
+        Function that reloads model if necesary and extracts trainable part 
+        of the model in preparation for target_task training. 
+        It only reloads model after the first task is trained, and only 
+        for finetuning. 
+
+        Parameters
+        -------------------
+        args: config.Param object 
+        ckpt_path: str
+        model: MultiTaskmodel object 
+        index: int 
+        strict: bool, 
+        task: Task object 
+
+        Returns 
+        -------------------
+        to_train: List of tuples of (name, weight) for trainable parameters
+
+    """
+    if index > 0 and args.transfer_paradigm == "finetune":
+        # Only reload after target_train for first task, and transfer_paradigm=finetune.
+        load_model_state(model, ckpt_path, args.cuda, skip_task_models=[task.name], strict=strict)
+
+    if args.transfer_paradigm == "frozen":
+        # will be empty if elmo = 0. scalar_mix_0 should always be
+        # pretrain scalars
+        elmo_scalars = [
+            (n, p)
+            for n, p in model.named_parameters()
+            if "scalar_mix" in n and "scalar_mix_0" not in n
+        ]
+        # Fails when sep_embs_for_skip is 0 and elmo_scalars has nonzero
+        # length.
+        assert_for_log(
+            not elmo_scalars or args.sep_embs_for_skip,
+            "Error: ELMo scalars loaded and will be updated in do_target_task_training but "
+            "they should not be updated! Check sep_embs_for_skip flag or make an issue.",
+        )
+
+    if args.transfer_paradigm == "finetune":
+        model_state = load_model_state(
```

Comment:
We already did this above.

Reply:
This is different - I realize it's a bit confusing so I'm going to collapse the above into the paradigm=frozen case.

Reply:
Ah, got it. I didn't realize that these were reused later.